src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,154 @@
+package apoc.atomic;
+
+import apoc.atomic.util.AtomicUtils;
+import apoc.periodic.Periodic;
+import apoc.util.ArrayBackedList;
+import apoc.util.MapUtil;
+import org.neo4j.graphdb.PropertyContainer;
+import org.neo4j.kernel.api.KernelTransaction;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+
+import java.lang.reflect.Array;
+import java.util.*;
+import java.util.stream.Stream;
+
+/**
+ * @author AgileLARUS
+ *
+ * @since 20-06-17
+ */
+public class Atomic {
+
+	@Context public GraphDatabaseAPI db;
+	@Context public Log               log;
+	@Context public KernelTransaction tx;
+
+	/**
+	 * increment a property's value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.add(node/relatonship,propertyName,number) Sums the property's value with the 'number' value ")
+	public Stream<AtomicResults> add(@Name("container") Object container, @Name("propertyName") String property, @Name("number") Number number) {
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Number oldValue = (Number) propertyContainer.getProperty(property);
```

Comment:
this is not atomic, we have to grab a (read)lock first thing, that's what the whole thing is about.